tap_yotpo/http.py

```diff
@@ -147,38 +148,31 @@ def authenticate(self):
             "client_secret": self.api_secret,
             "grant_type": GRANT_TYPE
         }
-
         request = requests.Request(method="POST", url=AUTH_URL, data=auth_body)
         response = self.prepare_and_send(request)
-        self.raise_for_error(response)
+        self.check_status(response, authentication_call=True)
         data = response.json()
         self._token = data['access_token']
 
     def GET(self, version, request_kwargs, *args, **kwargs):
         req = self.create_get_request(version, **request_kwargs)
         return self.request_with_handling(req, *args, **kwargs)
 
-    def raise_for_error(self, resp):
+    def check_status(self, response, authentication_call=False):
+        # Forming a response message for raising custom exception
         try:
-            resp.raise_for_status()
-        except (requests.HTTPError, requests.ConnectionError) as error:
-            try:
-                error_code = resp.status_code
-                # Forming a response message for raising custom exception
-                try:
-                    response_json = resp.json()
-                except Exception:
-                    response_json = {}
-
-                message = "HTTP-error-code: {}, Error: {}".format(
-                    error_code,
-                    response_json.get("status",ERROR_CODE_EXCEPTION_MAPPING.get(
-                        error_code, {})).get("message", "Unknown Error")
-                    )
-
-                exc = ERROR_CODE_EXCEPTION_MAPPING.get(
-                    error_code, {}).get("raise_exception", YotpoError)
-                raise exc(message, resp) from None
-
-            except (ValueError, TypeError):
-                raise YotpoError(error) from None
+            response_json = response.json()
+        except Exception:
+            response_json = {}
+        if response.status_code != 200:
+            message = "HTTP-error-code: {}, Error: {}".format(
+                response.status_code,
+                response_json.get("status", ERROR_CODE_EXCEPTION_MAPPING.get(
+                    response.status_code, {})).get("message", "Unknown Error")
+            )
+            exc = ERROR_CODE_EXCEPTION_MAPPING.get(
+                response.status_code, {}).get("raise_exception", YotpoError)
+            # Reauthenticating if it isn't raise while authentication and for all error except some known error
```

Comment:
@dbshah1212 Please update the comment as follows:
"Re-authenticating if got an authentication error while collecting stream data and response does not have Bad request (400), Forbidden (403) and Not found (404)

Reply:
Updated